modules/externalipam/ipam_validator.rb

```diff
@@ -0,0 +1,49 @@
+require 'proxy/validations'
+require 'externalipam/externalipam'
+require 'resolv'
+
+# Module containing validation methods for use by all External IPAM provider implementations
+module Proxy::Ipam::IpamValidator
+  include ::Proxy::Validations
+  include Proxy::Ipam::IpamHelper
+
+  def validate_required_params!(required_params, params)
+    err = []
+    required_params.each do |param|
+      unless params[param.to_sym]
+        err.push errors[param.to_sym]
+      end
+    end
+    raise Proxy::Validations::Error, err unless err.empty?
+  end
+
+  def validate_ip!(ip)
+    good_ip = ip =~ Regexp.union([Resolv::IPv4::Regex, Resolv::IPv6::Regex])
+    raise Proxy::Validations::Error, ERRORS[:bad_ip] if good_ip.nil?
+    ip
+  end
+
+  def validate_cidr!(address, prefix)
+    cidr = "#{address}/#{prefix}"
+    network = IPAddr.new(cidr).to_s
+    if IPAddr.new(cidr).to_s != IPAddr.new(address).to_s
```

Comment:
You can reuse the `network` variable:
```suggestion
    if network != IPAddr.new(address).to_s
```